soda/core/soda/execution/data_source.py

```diff
@@ -238,49 +238,71 @@ def sql_find_table_names(
     def sql_information_schema_identifier(self) -> str:
         return "information_schema.tables"
 
+    def sql_analyze_table(self, table: str) -> str | None:
+        return None
+
     ######################
     # Query Execution
     ######################
 
     def get_row_counts_all_tables(
-        self, include_tables: list[str] | None, exclude_tables: list[str] | None, query_name: str | None
+        self,
+        include_tables: list[str] | None = None,
+        exclude_tables: list[str] | None = None,
+        query_name: str | None = None,
     ) -> dict[str, int]:
         """
         Returns a dict that maps table names to row counts.
         """
         sql = self.sql_get_table_names_with_count(include_tables=include_tables, exclude_tables=exclude_tables)
         if sql:
             query = Query(
-                data_source_scan=self,
+                data_source_scan=self.data_source_scan,
                 unqualified_query_name=query_name or "get_row_counts_all_tables",
                 sql=sql,
             )
             query.execute()
             return {row[0]: row[1] for row in query.rows}
 
         # Single query to get the metadata not available, get the counts one by one.
-        all_tables = self.find_table_names()
+        all_tables = self.get_table_names(include_tables=include_tables, exclude_tables=exclude_tables)
         result = {}
 
         for table in all_tables:
             query = Query(
-                data_source_scan=self,
-                unqualified_query_name=f"get_row_count_{table}",
-                sql=self.sql_get_table_count(table),
+                data_source_scan=self.data_source_scan,
+                unqualified_query_name=f"{query_name}_{table}" or f"get_row_count_{table}",
```

Comment:
Didn't see this constructor string `or` string before
Wouldn't this always take the first?  Even if `query_name` or `table` are None ?

Reply:
ah this slipped, you are totally right, originally I used `{query_name} or f"get_row_count_{table}",` which would of course work, but then I added the _table part which broke it, thanks, fixed!